### ISSUE TITLE
[4.0] Uncaught TypeError: Cannot read property 'state' of undefined

### ISSUE THREAD

**Issue by amackintosh** [reactjs/redux#2715]
**Do you want to request a *feature* or report a *bug*?**  
Possible bug

**What is the current behavior?**
Detonation after install Redux 4.0

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar.**
1. Install Redux 4.0
2. Run app

**What is the expected behavior?**
Should work as normal :)

**Which versions of Redux, and which browser and OS are affected by this issue? Did this work in previous versions of Redux?**
Windows 10, Chrome newest
```
    "react": "^16.0.0",
    "react-apollo": "^1.4.16",
    "react-dom": "^16.0.0",
    "react-redux": "^5.0.6",
    "react-router-dom": "^4.2.2",
    "react-router-redux": "^5.0.0-alpha.8",
    "redux": "^4.0.0-beta.1",
    "redux-thunk": "^2.2.0",
```
It works correct with Redux 3.7.?

Here is the error:
![redux4](https://user-images.githubusercontent.com/25964567/32910521-575561fc-cabe-11e7-9d8b-77060fb6ba9b.png)

`./src/root.jsx`
``` javascript
import React from 'react'
import ReactDOM from 'react-dom'
import { ApolloClient, createNetworkInterface, ApolloProvider } from 'react-apollo'
import { withRouter, Route } from 'react-router-dom'
import { ConnectedRouter } from 'react-router-redux'
import browserHistory from './store/history'
import AuthGateway from './components/auth/AuthGateway'
import { AUTH_SUCCESS } from './components/auth/auth_types'
import App from './components/App'
import configure from './store'
import '../styles/app.scss'

const store = configure()

const tokenOnLoad = localStorage.getItem('token')
if (tokenOnLoad) store.dispatch({ type: AUTH_SUCCESS })

const networkInterface = createNetworkInterface({
  uri: 'http://localhost:8000/graphql',
  opts: { credentials: 'same-origin', mode: 'cors' },
})
networkInterface.use([{
  applyMiddleware(req, next) {
    if (!req.options.headers) req.options.headers = {}
    const token = localStorage.getItem('token')
    req.options.headers.authorization = token || null
    next()
  },
}])

const client = new ApolloClient({
  networkInterface,
  dataIdFromObject: o => o.id,
})

const Root = () => (
  <ApolloProvider store={store} client={client}>  // this is line 37 (in the above screenshot)
    <AuthGateway>
      <ConnectedRouter history={browserHistory}>
        <Route path="/" component={withRouter(App)} />
      </ConnectedRouter>
    </AuthGateway>
  </ApolloProvider>
)
ReactDOM.render(
  <Root />,
  document.querySelector('#root'),
)

export default client
```

**./src/store/index.js**
``` javascript
import { createStore, applyMiddleware } from 'redux'
import { routerMiddleware } from 'react-router-redux'
import reduxThunk from 'redux-thunk'
import tokenValidator from '../middleware/tokenvalidator'
import logger from '../middleware/logger'
import browserHistory from './history'
import rootReducer from '../reducers'

const historyMiddleware = routerMiddleware(browserHistory)

export default function configure(initialState) {
  const create = window.devToolsExtension
    ? window.devToolsExtension()(createStore)
    : createStore

  const createStoreWithMiddleware = applyMiddleware(
    reduxThunk,
    logger,
    tokenValidator,
    historyMiddleware,
  )(create)

  const store = createStoreWithMiddleware(rootReducer, initialState)
  return store
}
```

**./src/store/history.js**
``` javascript
import createHistory from 'history/createBrowserHistory'

export default createHistory()
```

I do not know what to make of this currently, but I do notice that it's related to the `store` and inside the store index file, it is calling `import { createStore, applyMiddleware } from 'redux'`.

I notice there were some changes to `applyMiddleware`, so something in my setup may point to a change there.

**Comment by timdorr** [comment]
Is this coming up as soon as you try to load the app?

**Comment by timdorr** [comment]
I think you're running into this issue: zalmoxisus/redux-devtools-extension#47

**Comment by amackintosh** [comment]
I think you are definitely correct. I omitted devtools extension, and it fired up without issue.

I tried to mess with it, but I couldn't get a combination to work.

I changed it from `window.devToolsExtention` to `window.__REDUX_DEVTOOLS_EXTENSION__`, because of https://github.com/zalmoxisus/redux-devtools-extension/issues/220

I would have to reorganize that whole store creation sequence if I was to try anything more than those quick stabs, which I will do if we are leaning towards issue on my end.

This post here makes me think it could be a quick change needed in Redux DevTools to support Redux 4: https://github.com/zalmoxisus/redux-devtools-extension/issues/220

Let me know if you want me to take any additional actions.
